bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/handler/channelhelper/EventChannelHelper.java

```diff
@@ -66,45 +67,39 @@ public void setNewData(@Nullable NAObject data) {
 
     @Override
     protected @Nullable State internalGetEvent(String channelId, Event event) {
-        switch (channelId) {
-            case CHANNEL_EVENT_TYPE:
-                return toStringType(event.getEventType());
-            case CHANNEL_EVENT_MESSAGE:
-                return toStringType(event.getName());
-            case CHANNEL_EVENT_TIME:
-                return new DateTimeType(event.getTime());
-            case CHANNEL_EVENT_PERSON_ID:
-                return toStringType(event.getPersonId());
-            case CHANNEL_EVENT_CAMERA_ID:
-                return toStringType(event.getCameraId());
-            case CHANNEL_EVENT_SUBTYPE:
-                return event.getSubTypeDescription().map(d -> toStringType(d)).orElse(UnDefType.NULL);
-            case CHANNEL_EVENT_SNAPSHOT:
-                return toRawType(event.getSnapshotUrl());
-            case CHANNEL_EVENT_SNAPSHOT_URL:
-                return toStringType(event.getSnapshotUrl());
-        }
-        return null;
+        return switch (channelId) {
+            case CHANNEL_EVENT_TYPE -> toStringType(event.getEventType());
+            case CHANNEL_EVENT_MESSAGE -> toStringType(event.getName());
+            case CHANNEL_EVENT_TIME -> new DateTimeType(event.getTime());
+            case CHANNEL_EVENT_PERSON_ID -> toStringType(event.getPersonId());
+            case CHANNEL_EVENT_CAMERA_ID -> toStringType(event.getCameraId());
+            case CHANNEL_EVENT_SUBTYPE ->
+                event.getSubTypeDescription().map(ChannelTypeUtils::toStringType).orElse(UnDefType.NULL);
+            case CHANNEL_EVENT_SNAPSHOT -> toRawType(event.getSnapshotUrl());
+            case CHANNEL_EVENT_SNAPSHOT_URL -> toStringType(event.getSnapshotUrl());
+            default -> null;
+        };
     }
 
     @Override
     protected @Nullable State internalGetHomeEvent(String channelId, @Nullable String groupId, HomeEvent event) {
-        switch (channelId) {
-            case CHANNEL_EVENT_VIDEO_STATUS:
-                return event.getVideoId() != null ? toStringType(event.getVideoStatus()) : UnDefType.NULL;
-            case CHANNEL_EVENT_VIDEO_LOCAL_URL:
-                return getStreamURL(true, event.getVideoId(), event.getVideoStatus());
-            case CHANNEL_EVENT_VIDEO_VPN_URL:
-                return getStreamURL(false, event.getVideoId(), event.getVideoStatus());
-        }
-        return null;
+        return switch (channelId) {
+            case CHANNEL_EVENT_VIDEO_STATUS ->
+                event.getVideoId() != null ? toStringType(event.getVideoStatus()) : UnDefType.NULL;
+            case CHANNEL_EVENT_VIDEO_LOCAL_URL -> getStreamURL(true, event.getVideoId(), event.getVideoStatus());
+            case CHANNEL_EVENT_VIDEO_VPN_URL -> getStreamURL(false, event.getVideoId(), event.getVideoStatus());
+            default -> null;
+        };
+    }
+
+    private String getUrl(boolean local) {
+        return Objects.requireNonNull(local ? localUrl : vpnUrl);
```

Comment:
Same question as in the other PR: how can you be sure that `localUrl` is not null ?